### PR TITLE
Add skin support for hit judgements

### PR DESCRIPTION
**Pull request by peppy**
Also makes sizing of skinned drawables more flexible.

### CODE REVIEW

File: osu.Game/Rulesets/Judgements/DrawableJudgement.cs [hunk 1]
```diff
@@ -30,31 +34,37 @@ public DrawableJudgement(Judgement judgement)
...
+                    legacyName = "hit0";
+                    break;
+                case HitResult.Meh:
+                    legacyName = "hit50";
+                    break;
+                case HitResult.Good:
+                    legacyName = "hit100";
+                    break;
+                case HitResult.Great:
+                    legacyName = "hit300";
```
Comment (smoogipoo) [inline osu.Game/Rulesets/Judgements/DrawableJudgement.cs]: I don't like that we're polluting osu.Game with this stuff. This should be done individually for each legacy ruleset imo.

File: osu.Game/Rulesets/Judgements/DrawableJudgement.cs [hunk 2]
```diff
@@ -30,31 +34,37 @@ public DrawableJudgement(Judgement judgement)
...
                     break;
             }
+
+            Child = new SkinnableDrawable($"Play/osu/{legacyName}", _ => JudgementText = new OsuSpriteText
```
Comment (smoogipoo) [inline osu.Game/Rulesets/Judgements/DrawableJudgement.cs]: How about instead exposing a `virtual string GetResourceName(HitResult result) => "Play/osu/{result.ToString().ToLower()";`? Two reasons:
1. Eventually we're not going to have `Play/osu/`. When we move these resources to each gamemode they should exist as simply `Play/`.
2. Will allow us to override this lookup further depending on additional construction parameters (eg, per-key judgements in mania).

File: osu.Game/Skinning/LegacySkin.cs [hunk 3]
```diff
@@ -48,7 +43,8 @@ private class LegacySkinResourceStore : IResourceStore<byte[]>
             private readonly IResourceStore<byte[]> underlyingStore;
 
             private string getPathForFile(string filename) =>
-                skin.Files.FirstOrDefault(f => string.Equals(Path.GetFileNameWithoutExtension(f.Filename), filename.Split('/').Last(), StringComparison.InvariantCultureIgnoreCase))?.FileInfo.StoragePath;
+                skin.Files.FirstOrDefault(f => string.Equals(Path.GetFileNameWithoutExtension(f.Filename), filename.Split('/').Last(), StringComparison.InvariantCultureIgnoreCase))?.FileInfo
+                    .StoragePath;
```
Comment (smoogipoo) [inline osu.Game/Skinning/LegacySkin.cs]: Put this back on the same line.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

**Comment by smoogipoo** [comment]
screen shot 2018-03-13 at 9 55 57 am

Known issue? The text size is too small.

**Comment by peppy** [comment]
Text size fixed via ppy/osu-framework#1459.

**Review by smoogipoo — APPROVED** [review]
looks ok to me

**Pull request merged**